cassandra/cluster.py

```diff
@@ -5050,6 +5054,8 @@ def _retry_task(self, reuse_connection, host):
         if reuse_connection and self._query(host) is not None:
             return
 
+        if delay:
+            time.sleep(delay)
```

Comment:
I don't like sleeps, I fear it will result in more problems where there are no threads in executor to do something, so the driver hangs.
I think we have some method to schedule something with delay. Maybe we could use that instead?

Reply:
ThreadPoolExecutor doesn't have scheduling out of the  box
and if it would have we need to align to all of the backends.

anyhow this is how retires works toady, and if running out of workers, it can happen today, even quicker.
and it won't get stuck, it's just gonna wait in line.

I still think it's better than forcing the users to do their own retry logic.

Reply:
I've found that we do have a schedule implementation, and I've change it to use it.
we'll need to keep in mind there's only one queue for that scheduler